app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -136,11 +139,21 @@ class SettingsFragment : PreferenceFragmentCompat() {
     }
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
-        val preferencesId = if (FeatureFlags.newIconSet) {
-            R.xml.preferences_without_icons
+
+        val preferencesId = if (isExperimentBranch()) {
+            if (FeatureFlags.newIconSet) {
+                R.xml.preferences_without_icons_experiment
+            } else {
+                R.xml.preferences_experiment
```

Comment:
Having the extra `XML` files make modification a bit harder as we will have to make change in 4 files, but I can't think on better way to do it, without making things a bit complex, and as these changes will be momentarily I guess it's OK. Maybe we could extract the getting the preference ID into a function and add some comments for people to be aware that changes to the setting `XML` could required changes on the 4 `XML` files.